### DISCUSSION TITLE
[Feature] Estimate location based on reverse image search

### DISCUSSION THREAD

**Discussion by petersuba** [immich-app/immich#10881] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

In my archive I have pictures where I am not sure where exactly they are from (e.g. I may know the country or even city, but don't remember where exactly), and they were taken without GPS coordinates (i.e. they are old and/or not taken with a smartphone) but they contain prominent features. As a result they are now in the immich database without location.

Currently the way to find and tag an approximate location I do is to save the photo, go to Google image search, find images containing the same prominent feature, extract the location from those pictures and use that to manually navigate on the map and tag the original picture.

A much faster way to achieve the same would be to create a utility/integration directly from within immich, using the Google search API (or alternatively, a similar service). Given that this isn't a foolproof method, this is not something that can be automated across many photos so this would be an interactive method/dialog that one can go through and select the possible matches.

One option is a Utility similar to handling duplicates, i.e. these would be for Geotagging pictures. The UI could cycle through only the untagged photos. Each one the UI could present the photo to be tagged, alongside possible usable similar photos from image searches (i.e. the UI would only show pictures found that have possible locations available with them, either through EXIF location or GPS coordinate, and if there are multiple possible approximate locations, it would show one per approximate location with a map and marker each e.g. 3 max.) The user could then adjust the marker on the map and click "tag as location" or similar.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#9501
